ssvsigner/ekm/remote_key_manager.go

```diff
@@ -116,21 +116,65 @@ func (km *RemoteKeyManager) AddShare(
 		PubKey:           pubKey,
 	}
 
-	if err := km.signerClient.AddValidators(ctx, shareKeys); err != nil {
+	statuses, err := km.signerClient.AddValidators(ctx, shareKeys)
+	if err != nil {
 		return fmt.Errorf("add validator: %w", err)
 	}
 
+	for _, status := range statuses {
+		switch status {
+		case web3signer.StatusImported:
+
+		case web3signer.StatusDuplicated:
+			// A failed request does not guarantee that the keys were not added.
+			// It's possible that the ssv-signer successfully added the keys,
+			// but a network error occurred before a response could be received.
+			// Or, if ssv-signer is behind a load balancer, the load balancer may return an error.
+			// In such cases, the node would crash and, upon restarting, encounter a duplicate key error.
+			// To handle this gracefully, we allow returning a duplicate key error without treating it as a failure.
+			km.logger.Warn("Attempted to add already existing share to the remote signer. " +
+				"This is expected in the first block after failed sync")
+
+		default:
+			return fmt.Errorf("unexpected status %s", status)
+		}
+	}
+
+	if err := km.BumpSlashingProtection(pubKey); err != nil {
+		return fmt.Errorf("could not bump slashing protection: %w", err)
+	}
```

Comment:
This is a critical bug - `BumpSlashingProtection` is called twice.

Reply:
I've opened issue #2332  to add regression tests ensuring BumpSlashingProtection is never called twice. This will prevent the bug from being reintroduced in future refactoring.